### PR TITLE
feat(systemd-timedated): introducing the systemd-timedated module

### PR DESCRIPTION
**Pull request by johannbg**
Introducing the systemd-timedated module.
systemd-timesyncd will be introduced as a separated module.

This pull request changes...

## Checklist
- [ ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #

### CODE REVIEW

**Comment by haraldh** [comment]
needs rebase for dracut.spec

File: modules.d/01systemd-timedated/module-setup.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# This file is part of dracut.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Prerequisite check(s) for module.
+check() {
+
+    # If the binary(s) requirements are not fulfilled the module can't be installed
+    require_binaries timedatectl || return 1
+    require_binaries "$systemdutildir"/systemd-timedated || return 1
```
Comment (haraldh) [inline modules.d/01systemd-timedated/module-setup.sh:10]:
```suggestion
    require_binaries \
        timedatectl \
        "$systemdutildir"/systemd-timedated \
        || return 1
```

File: dracut.spec [hunk 2]
```diff
@@ -324,6 +324,7 @@ echo 'dracut_rescue_image="yes"' > $RPM_BUILD_ROOT%{dracutlibdir}/dracut.conf.d/
 %endif
 %{dracutlibdir}/modules.d/01systemd-ac-power
 %{dracutlibdir}/modules.d/01systemd-ask-password
+%{dracutlibdir}/modules.d/01systemd-timedated
```
Comment (haraldh) [inline dracut.spec:327]: I thought it was alphabetically sorted :)

**Review by haraldh — CHANGES_REQUESTED** [review]

File: dracut.spec (hunk 2 above)
Comment (johannbg) [reply dracut.spec]: Was is correct and now is again :smile:

**Pull request merged**